### ISSUE TITLE
Alert dialog closes before asynchronous form submission

### ISSUE THREAD

**Issue by leemorrowdev** [radix-ui/primitives#1779]
### Current Behavior

When using `<AlertDialog.Action>` on form submission, the alert dialog is unmounted immediately instead of on `setOpen(false)`. This results in the following warning in the console:

```
Form submission canceled because the form is not connected
```

### Expected behavior

The alert dialog remains open until `setOpen(false)` runs after the promise resolves

### Reproducible example

[CodeSandbox Template](https://codesandbox.io/s/quizzical-lalande-xk6mi8?file=/src/App.js)

### Additional context

The code in the reproducible example was [derived from the documentation](https://www.radix-ui.com/docs/primitives/components/alert-dialog#close-after-asynchronous-form-submission)

### Your environment

| Software         | Name(s) | Version |
| ---------------- | ------- | ------- |
| Radix Package(s) |    @radix-ui/react-dialog     |   1.0.2    |
| React            | n/a     |    18.2.0    |
| Browser          |    Chrome     |    107.0.5304.107     |
| Assistive tech   |         |         |
| Node             | n/a     |         |
| npm/yarn         |    npm     |    8.5.0     |
| Operating System |    Windows 10     |     21H1    |

**Comment by leemorrowdev** [comment]
Hi,

Sorry for the confusion, I must be missing something here. I based my reproducible example on [the asynchronous form submission example in the documentation](https://www.radix-ui.com/docs/primitives/components/alert-dialog#close-after-asynchronous-form-submission) and I think I am doing that?

Thanks!

**Comment by benoitgrelard** [comment]
Ah yes sorry, the example is slightly wrong, we are aware of it (I'll fix it today).
It's missing `event.preventDefault` in `onSubmit` and also it should use your own submit button instead of `Action` to submit.

**Comment by andy-hook** [comment]
@benoitgrelard I was about to follow up here as well. I think the ergonomics of `Action` and `Close` is especially confusing in this use case since `preventDefault` inside `onClick` prevents the `onSubmit` handler.

@leemorrowdev  Here is a working example if it's useful: https://codesandbox.io/s/alert-dialog-form-submission-async-d0022g?file=/src/App.js

**Comment by benoitgrelard** [comment]
@andy-hook exactly, I'm going to update the examples today as it's a few people who have noted it's not working correctly 👍

**Comment by benoitgrelard** [comment]
https://github.com/radix-ui/website/pull/544
